### PR TITLE
Fix/hck 4555 comment deactivated relationships

### PR DESCRIPTION
**Pull request by ArzamastsevVladyslav**

### CODE REVIEW

File: forward_engineering/utils/general.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+/**
```
Comment (chulanovskyi-bs) [inline forward_engineering/utils/general.js:1]: we already have `forward_engineering/helpers/utils.js`, could it be placed at least under the "helpers" folder, and have more explicit name, something like `statementsJoiner.js`?

File: forward_engineering/ddlProvider.js [hunk 2]
```diff
@@ -102,20 +104,40 @@ module.exports = (baseProvider, options, app) => {
...
+			const statementDtosToAddInsideTable = [
+				...columnStatementDtos,
+				compositePrimaryKeyOutlineConstraintStatementDto,
+				...foreignKeyConstraintStatementDtos,
+			]
+				.filter(statementDto => Boolean(statementDto.statement));
+			const joinedStatementsToAddInsideTable = joinActivatedAndDeactivatedStatements({
+				statementDtos: statementDtosToAddInsideTable,
+				delimiter: ',\n',
+				delimiterForLastActivatedStatement: '\n',
```
Comment (chulanovskyi-bs) [inline forward_engineering/ddlProvider.js:133]: these args value are already being declared as a [default here](https://github.com/hackolade/BigQuery/pull/45/files#diff-e31abefe4112fbf65021241569c69f0adf6a22ad5a448437d552790ac68272beR13-R14)

File: forward_engineering/utils/general.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+ *     isActivated: boolean,
+ * }>}
+ * @param delimiter {string | undefined}
+ * @param delimiterForLastActivatedStatement {string | undefined}
+ * @return {string}
+ * */
+const joinActivatedAndDeactivatedStatements = ({
+                                                   statementDtos,
+                                                   delimiter = ',\n',
+                                                   delimiterForLastActivatedStatement = '\n',
```
Comment (chulanovskyi-bs) [inline forward_engineering/utils/general.js:58]: if you are going to use these params only within `getDelimiterForJoiningStatementsBasedOnActivation` [function](https://github.com/hackolade/BigQuery/pull/45/files#diff-e31abefe4112fbf65021241569c69f0adf6a22ad5a448437d552790ac68272beR13-R14), then I can't see the reason providing a default values here

File: forward_engineering/utils/general.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                                               }) => {
+    const partsOfFinalStatement = [];
+    const lastIndexOfActivatedStatement = getLastIndexOfActivatedStatement(statementDtos);
+
+    for (let i = 0; i < statementDtos.length; i++) {
```
Comment (chulanovskyi-bs) [inline forward_engineering/utils/general.js:63]: the `.map` have a built-in second arg as `index`. Can easily replace this loop with a functional approach, and eliminate the `const partsOfFinalStatement = [];` and eventual pushing the items manually 👍🏻

File: forward_engineering/utils/general.js (hunk 1 above)
Comment (ArzamastsevVladyslav) [reply forward_engineering/utils/general.js:1]: We started implementing a new plugin structure. There must be only 1 utility file with functions that used by FE and Alter script, and it is `general/utils.js`

File: forward_engineering/utils/general.js (hunk 3 above)
Comment (ArzamastsevVladyslav) [reply forward_engineering/utils/general.js:58]: Spreading default value declarations throughout several functions is a bad design + hard to read

File: forward_engineering/utils/general.js (hunk 4 above)
Comment (ArzamastsevVladyslav) [reply forward_engineering/utils/general.js:63]: We will be replacing it with `flatMap` which is harder to read and understand

File: forward_engineering/utils/general.js (hunk 1 above)
Comment (chulanovskyi-bs) [reply forward_engineering/utils/general.js:1]: > We started implementing a new plugin structure
Where can I check the agreed approach with that 1 file limitation?)

Comment (ArzamastsevVladyslav) [reply forward_engineering/utils/general.js:1]: https://app.gitbook.com/o/HBtg1gLTy0nw4NaX0MaV/s/RsuO9R6FbfIJtofdqr4L/~/changes/213/guides-and-how-to/plugins/alter-and-fe-script.-directory-structure-and-general-flow

**Review by chulanovskyi-bs — APPROVED** [review]

File: forward_engineering/utils/statementJoiner.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+/**
+ * @param statementDtos {Array<{
+ *     statement: string,
+ *     isActivated: boolean,
+ * }>}
+ * @return {number}
+ */
+const getLastIndexOfActivatedStatement = (statementDtos) => {
```
Comment (VitaliiBedletskyi) [inline forward_engineering/utils/statementJoiner.js:35]: Destructured property bags)

**Review by VitaliiBedletskyi — CHANGES_REQUESTED** [review]

**Review by VitaliiBedletskyi — APPROVED** [review]

**Pull request merged**